src/main/java/jdwp/JDWP.java

```diff
@@ -65,6 +76,15 @@ static int getNewTokenId() {
         return count;
     }
 
+    static long getNewAsmId() {
+        long count = --asmIdCounter;
+        // If we ever wrap around.
+        if (count >= 0) {
```

Comment:
One small bug I realized: should be `if (count == JDWP.optimizedVarID)` since `JDWP.optimizedVarID = -Long.MAX_VALUE` and so  `-Long.MAX_VALUE` is already taken (but also doubting asmID will ever reach that far)